eng/test.yaml

```diff
@@ -0,0 +1,67 @@
+steps: 
+
+- task: DotNetCoreCLI@2
+  displayName: Restore (dotnet restore)
+  inputs:
+    command: 'custom'
+    projects: |
+      src/**/*.Tests.csproj
+      !src/**/Microsoft.DotNet.Darc.Tests.csproj
+    custom: 'restore'
+  condition: succeededOrFailed()
+
+- task: DotNetCoreCLI@2
```

Comment:
Is this going to run the unit tests a second time?  Today the unit tests are running as part of the build (though some of them were missing from the solution, which you fixed here).

Reply:
It will run them again to capture code coverage and report on the results, however, it does not rebuild the projects.

Reply:
I'll go through the existing test results and make sure they get wrapped up in this step so there aren't duplicate results being reported.

Reply:
Removed the upload of test results in this yaml since Arcade is handling that already for Arcade Services (may explore incorporating code coverage in Arcade in another issue).

Reply:
Can't we what we need to capture for code coverage in the first run of the tests instead?

Reply:
@riarenas I briefly explored that and it would require making changes to cibuild.cmd and whatever that calls, which I don't believe is `dotnet test`. If that was the route we'd choose to go down, then there would be a disparity in how code coverage is collected between Helix and Arcade Services, and probably OS Onboarding as well. I chatted with @mmitche  about introducing code coverage into Arcade itself, but that work is outside the scope of this epic. (Apparently, it's an asked for feature, so it's not like it would be unwanted)

Reply:
Created https://github.com/dotnet/arcade/issues/4551 to refactor this code coverage collection so we're not duplicating testing work.